tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -0,0 +1,379 @@
+  const { act, data } = useBackend<ManufacturerData>(context);
+  const [repeat, toggleRepeatVar] = useSharedState(context, "repeat", data.repeat);
+  const [speed, setSpeedVar] = useSharedState(context, "speed", data.speed);
+  let toggleRepeat = () => {
+    act("repeat");
+    toggleRepeatVar(!repeat);
+  };
+  let updateSpeed = (newValue:number) => {
+    act("speed", { "value": newValue });
+    setSpeedVar(newValue);
+  };
+  let usable_blueprints = data.available_blueprints;
+  let dropdowns = [];
+  for (let i of data.all_categories) {
+    if (!usable_blueprints[i]) continue;
+    dropdowns.push(<CategoryDropdown category={i} blueprints={usable_blueprints[i]} />);
+  }
+  return (
+    <Window width={1200} height={600} title={data.fabricator_name}>
+      <Window.Content>
+        <Flex>
+          <Flex.Item
+            style={{ "width": "80%" }}
+            mr={1}
+          >
+            <Section>
+              {dropdowns}
+            </Section>
+          </Flex.Item>
+          <Flex.Item
+            style={{ "width": "20%" }}
+          >
+            <Section
+              fill
+              align="center"
+            >
+              <Input placeholder={"Search..."} icon={"search"} style={{ "width": "100%" }} />
+              <Section title="Materials Loaded" style={{ "width": "100%" }} textAlign="center" pt={1} backgroundColor={backgroundPop}>
+                <LoadedMaterials resources={data.resources} />
+              </Section>
+              <Slider
+                my={2}
+                minValue={1}
+                value={speed}
+                maxValue={3}
+                step={1}
+                stepPixelSize={100}
+                onChange={(_e:any, value:number) => updateSpeed(value)}
+              >
+                Speed: {speed}
+              </Slider>
+              <CardInfo />
+              <Flex
+                backgroundColor={backgroundPop}
+                my={1}
+                py={1}
+                style={{ "align-items": "center" }}
+              >
+                <Flex.Item grow>
+                  Repeat: {repeat ? "On" : "Off"}
+                </Flex.Item>
+                <Flex.Item>
+                  <Button icon="repeat" onClick={() => toggleRepeat()}>Toggle Repeat</Button>
+                </Flex.Item>
+              </Flex>
+
+              <CollapsibleWireMenu />
+
+              {data.rockboxes.map((rockbox:Rockbox) => (
+                <Section
+                  key={rockbox.name}
+                  backgroundColor={backgroundPop}
+                  title={rockbox.name+ " at "+rockbox.area_name}
+                  fontSize={0.875}
+                >
+                  {rockbox.ores.map((ore:Ore) => (
+                    <Button
+                      fontSize={1}
+                      key={ore.name}
+                      onClick={() => act("ore_purchase", { "ore": ore.name, "storage_ref": rockbox.reference })}
+                      style={{ "width": "90%" }}
```

Comment:
`fluid` instead of setting width, if you're trying to have them fill the whole thing (90% is functionally the whole thing).

Reply:
done, kind of jumping around with these code comments but i basically deleted all the custom styling and started from scratch. lead me down an interesting rabbit hole as well